include/jit/jit_utils.h

```diff
@@ -40,42 +40,29 @@ struct has_data_ptr : std::false_type {};
 template <typename T>
 struct has_data_ptr<
     T,
-    std::enable_if_t<std::conjunction_v<
-        std::is_same<decltype(std::declval<std::remove_reference_t<T>>().data_ptr()), void *>,
-        std::is_same<decltype(std::declval<std::remove_reference_t<T>>().scalar_type()), c10::ScalarType>>>>
+    std::void_t<decltype(std::declval<T>().data_ptr()), decltype(std::declval<T>().scalar_type())>>
     : std::true_type {};
 
 template <typename T>
-struct triton_type_helper {};
+struct triton_type_helper;
 
-template <>
-struct triton_type_helper<bool> {
-  static constexpr const char *name = "i1";
-};
-template <>
-struct triton_type_helper<int> {
-  static constexpr const char *name = "i32";
-};
-template <>
-struct triton_type_helper<int64_t> {
-  static constexpr const char *name = "i64";
-};
+#define DEFINE_TRITON_TYPE(T, Name)           \
+  template <>                                 \
+  struct triton_type_helper<T> {              \
+    static constexpr const char *name = Name; \
+  };
 
-template <>
-struct triton_type_helper<float> {
-  static constexpr const char *name = "f32";
-};
-template <>
-struct triton_type_helper<double> {
-  static constexpr const char *name = "f64";
-};
-template <>
-struct triton_type_helper<std::nullptr_t> {
-  static constexpr const char *name = "*i8";
-};
+DEFINE_TRITON_TYPE(bool, "i1")
```

Comment:
I prefer define the macro with ";" and use it with ";". This would have similar semantics as normal statements.